libs/os/dir.cpp

```diff
@@ -7,34 +7,28 @@
 
 Directory::Directory(const char* name) {
     this->dir = new QDir(name);
+    QStringList entryList = this->dir->entryList(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot);
+    for(int i = 0; i < entryList.size(); i++) {
+        QString file_path = entryList.at(i);
+        QString file_name = QFileInfo(file_path).fileName();
+        this->entries.push_back(file_name.toLatin1().data());
```

Comment:
You store pointer to temp string here. Do you even test? This is straightforward crash.